plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergUtil.java

```diff
@@ -38,18 +48,86 @@ public static TableInput getTableInput(String tableName, Optional<String> owner,
     {
         ImmutableMap.Builder<String, String> tableParameters = ImmutableMap.<String, String>builder()
                 .putAll(parameters)
-                .put(METADATA_LOCATION_PROP, metadataLocation);
+                .put(METADATA_LOCATION_PROP, metadataLocation)
+                .put(COLUMNS_INFO_VALID_FOR_METADATA_LOCATION, metadataLocation);
         if (!parameters.containsKey(TABLE_TYPE_PROP)) {
             tableParameters.put(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE.toUpperCase(ENGLISH));
         }
         return new TableInput()
                 .withName(tableName)
                 .withOwner(owner.orElse(null))
                 .withParameters(tableParameters.buildOrThrow())
+                .withStorageDescriptor(new StorageDescriptor()
+                        .withColumns(columns(metadata)))
                 // Iceberg does not distinguish managed and external tables, all tables are treated the same and marked as EXTERNAL
                 .withTableType(EXTERNAL_TABLE.name());
     }
 
+    private static List<Column> columns(TableMetadata metadata)
+    {
+        return metadata.schema().columns().stream()
+                .map(field -> {
+                    String glueTypeString = toGlueTypeStringLossy(field.type());
+                    String icebergTypeString = field.type().toString();
+                    Column column = new Column()
+                            .withName(field.name())
+                            .withType(glueTypeString)
+                            .withComment(field.doc());
+                    if (!glueTypeString.equals(icebergTypeString)) {
+                        column.setParameters(ImmutableMap.of(COLUMN_ICEBERG_TYPE_PROPERTY, icebergTypeString));
```

Comment:
The idea is to avoid storing unnecessary property, if it happens to be unnecessary
We may need a limit for case when property value goes beyond limit. (there must be a limit)